src/worker.js

```diff
@@ -81,7 +81,10 @@ else if (typeof process !== 'undefined') {
     var parentPort  = WorkerThreads.parentPort;
     worker.send = parentPort.postMessage.bind(parentPort);
     worker.on = parentPort.on.bind(parentPort);
-    worker.exit = process.exit.bind(process);
+    worker.exit = function (code) {
+      process.exitCode = code;
+      parentPort.postMessage('__workerpool-terminate__');
```

Comment:
I am confused on why we are seeing a message to the `parent` for shutdown when this message typically sent from parent to child, not child to parent. The parent message handler does not have implementations to deal with this message type.

Perhaps I am missing something, but to me this is not going to achieve the goals of providing the exit code on shutdown.

Reply:
Yes, I have the same question indeed, this message is only send from the main process to the worker right now. 

I would expect something like:

```js
worker.exit = function (code) {
  process.exitCode = code
  process.exit()
}
```